zebra-network/src/peer_set/initialize.rs

```diff
@@ -256,7 +270,29 @@ where
     let mut handshakes = FuturesUnordered::new();
     // TODO: replace with *success_count_tx.borrow() in Tokio 1.6
     let mut success_count = 0;
-    for candidate in initial_meta_addr {
+    for addr in initial_peers {
+        // 0. Update the state of this peer in the address book
+        //
+        // Note: these address book updates are sent to a channel, so they might be
+        // applied after updates from concurrent tasks.
+        let seeder_meta_addr = MetaAddr::new_seed_meta_addr(addr);
+        let _ = timestamp_collector
+            .send(MetaAddr::new_seed_change(seeder_meta_addr))
+            .await;
+
+        // If we've had enough successes, don't spawn any more tasks
+        if success_count >= peerset_initial_target_size {
+            continue;
+        }
+
+        // Otherwise, send an attempt to the address book
+        let update_change = MetaAddr::update_attempt(addr);
+        let _ = timestamp_collector.send(update_change).await;
+        // Apply the change, just like the AddressBook would
+        let candidate = update_change
+            .into_meta_addr(Some(seeder_meta_addr))
+            .expect("unexpected invalid seeder to attempt transition");
```

Comment:
Update the state of this peer in the address book
- Send `NeverAttemptedSeed` for each initial peer
- If we've had enough successes, just dump all the remaining peers in the address book as `NeverAttemptedSeed`
- Otherwise, send an `AttemptPending` to the address book
- And spawn a task for the handshake

Reply:
We could be filling up this channel and hanging or deadlocking?
The sleep between peers should help with that, but it would be nice to find the root cause.

Reply:
TODO: split off any extra channel usage into a post-MetaAddr-refactor PR